### ISSUE TITLE
Web - Workspace - Delay when navigate to Workspace page

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#33459]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.15-1
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. Navigate to staging.new.expensify.com
2. Go to Settings> Workspace

## Expected Result:
Transition from Settings to Workspace page should be smooth and there should be no delays

## Actual Result:
Delay when navigate to Workspace page

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] Windows: Chrome 
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/6e47ba49-ae81-4fcb-95b9-cfed3fcaf270

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by puneetlath** [comment]
@jjcoffee thoughts?

**Comment by jjcoffee** [comment]
Sorry I was OOO, reviewing today!

**Comment by jjcoffee** [comment]
Hmm strange that this would be happening again after it apparently being fixed in https://github.com/Expensify/App/issues/27611. I'd like to see that addressed in your proposal if possible @ZhenjaHorbach.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by jjcoffee** [comment]
Friendly bump @ZhenjaHorbach on [this comment](https://github.com/Expensify/App/issues/33459#issuecomment-1874132626).

**Comment by jjcoffee** [comment]
I can still repro on Linux Chrome but not on MacOS. The issue states that this happens on Windows, so I'd imagine it's an issue that's affecting non-Mac OSes.

**Comment by jjcoffee** [comment]
Still waiting for proposals. Similar issue here https://github.com/Expensify/App/issues/34141.

**Comment by jjcoffee** [comment]
@neonbhai @ZhenjaHorbach Would you be able to update your proposals with a demo of your fix in Chrome on Windows or Linux? I'm unsure if delaying the start of the Lottie animation until the modal animation completes will end up looking a bit too janky.

**Comment by jjcoffee** [comment]
Friendly bump @neonbhai @ZhenjaHorbach! Otherwise waiting for proposals.

**Comment by jjcoffee** [comment]
@samilabud You have a [proposal](https://github.com/Expensify/App/issues/34141#issuecomment-1883332378) on this related issue https://github.com/Expensify/App/issues/34141, would you mind reposting it here and verifying that it works for this issue too?

**Comment by samilabud** [comment]
> @samilabud You have a [proposal](https://github.com/Expensify/App/issues/34141#issuecomment-1883332378) on this related issue #34141, would you mind reposting it here and verifying that it works for this issue too? Or maybe there's a way to generalise your solution to work here too.


Thanks for reaching out, I appreciate it!

## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Workspace - Delay when navigate to Workspace page

### What is the root cause of that problem?
The root cause is a combination of react native navigation transition and lootie animation rendering, [the current one](https://github.com/Expensify/App/blob/41f051b64f0e0f0792813f1fd3f881ddb194f7f9/src/components/LottieAnimations/index.tsx#L34) is not optimized

### What changes do you think we should make in order to solve the problem?
We can get the original one and fix it in [lootiefiles](https://lottiefiles.com/), or change the current one for another one approved by the UI/UI Team.

Here is how it looks with another lootie file which is optimized:

https://github.com/Expensify/App/assets/5216036/93657b48-9875-435b-82a6-8728e9d651b7

Before with not optimized:

https://github.com/Expensify/App/assets/5216036/a1f84a9f-cbe5-4e40-9c19-727dbf5a690c

**Comment by jjcoffee** [comment]
@samilabud Interesting! So it's a different fix for this issue from the "related" one? If so, I'm still curious why this only affects Windows (and Linux, but we don't support that so fine to focus on Windows).

**Comment by samilabud** [comment]
> @samilabud Interesting! So it's a different fix for this issue from the "[related](https://github.com/Expensify/App/issues/34141)" one? If so, I'm still curious why this only affects Windows (and Linux, but we don't support that so fine to focus on Windows).

Yes, it is very interesting, in fact the previous proposal ([the releated one](https://github.com/Expensify/App/issues/34141)) is already implemented here, I mean the function is already memoized.

After seeing that I was checking and removed everything except the Lottie animation and it still does the same thing except when I remove the Lottie file and keep everything else, for these reasons I tried with different Lottie files and only the current one causes the delay.

BTW in both scenarios the common denominator is the navigation animation, it seems that the bug occurs when a component renders at the same time as the navigation animation occurs, I mean the first scenario was fixed because it was memoized (not re-rendered) and this one here occurs because the lootie animation renders at the same time as the navigation animation occurs (when the lootie animation has few or no rerenders this does not occur).

**Comment by jjcoffee** [comment]
Thanks @samilabud, have you tested optimising the file as you suggested in your proposal?

**Comment by samilabud** [comment]
> Thanks @samilabud, have you tested optimising the file as you suggested in your proposal?

I would have liked to but the Lootie file is compiled and can not be edited (I would need the original one), for that reason I tried with the others that were available there.

**Comment by jjcoffee** [comment]
This is no longer an issue since the Ideal Nav changes got merged. We can close @puneetlath :bow:

**Comment by puneetlath** [comment]
Ah ok, thanks for letting me know!
